gwctl/pkg/types/params.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"testing"
+
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/dynamic"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"sigs.k8s.io/gateway-api/gwctl/pkg/common"
+	"sigs.k8s.io/gateway-api/gwctl/pkg/policymanager"
+)
+
+type Params struct {
```

Comment:
Nit: I'm not really sure what the best name for this is, but `Params` doesn't seem quite right. It seems like this is a mix of 3 k8s clients, an internal data structure, and an io writer. Are there other ways we can split this up? How broadly is this type used? Are there common combinations that only need a subset of what's in here?

Reply:
This is definitely a valid commment. I decided to not remain blocked on this initial friction and start with something simple like Params to get moving.
- At the moment, this type is kinda used like a global "context" which is consumed by majority of the functions.
- Initially, I just had three K8s clients here (and hence I initially called it Clients). But as things evolved, and now we have two more things.
- Similar to your other suggestion about `describeViews`, this too definitely requires some code refactor.

I would completely understand the point of view to hold this PR until we fix this (Just say so without any further explanation and I'll be glad to address it here). But I was planning on taking this up separately as part of some code refactoring which is focused. I'll need some time to think the solution here.

Reply:
Refactored. Now we have a `CmdParams` but that's limited in scope to only being required by the CLI interface (and not need when library packages are used)